cohort-engine/src/main/java/com/ibm/cohort/engine/InsecureApacheRestfulClientFactory.java

```diff
@@ -0,0 +1,186 @@
+	 * Constructor
+	 *
+	 * @param theContext
+	 *            The context
+	 */
+	public InsecureApacheRestfulClientFactory(FhirContext theContext) {
+		super(theContext);
+	}
+
+	@Override
+	protected synchronized ApacheHttpClient getHttpClient(String theServerBase) {
+		return new ApacheHttpClient(getNativeHttpClient(), new StringBuilder(theServerBase), null, null, null, null);
+	}
+
+	@Override
+	public synchronized IHttpClient getHttpClient(StringBuilder theUrl, Map<String, List<String>> theIfNoneExistParams,
+												  String theIfNoneExistString, RequestTypeEnum theRequestType, List<Header> theHeaders) {
+		return new ApacheHttpClient(getNativeHttpClient(), theUrl, theIfNoneExistParams, theIfNoneExistString, theRequestType,
+				theHeaders);
+	}
+
+	public HttpClient getNativeHttpClient() {
+		if (myHttpClient == null) {
+
+			PoolingHttpClientConnectionManager connectionManager = createInsecureHttpConnectionManager(5000, TimeUnit.MILLISECONDS);
+			connectionManager.setMaxTotal(getPoolMaxTotal());
+			connectionManager.setDefaultMaxPerRoute(getPoolMaxPerRoute());
+
+			//TODO: Use of a deprecated method should be resolved.
+			RequestConfig defaultRequestConfig =
+					RequestConfig.custom()
+							.setSocketTimeout(getSocketTimeout())
+							.setConnectTimeout(getConnectTimeout())
+							.setConnectionRequestTimeout(getConnectionRequestTimeout())
+							.setStaleConnectionCheckEnabled(true)
+							.setProxy(myProxy)
+							.build();
+
+			HttpClientBuilder builder = HttpClients.custom().setConnectionManager(connectionManager)
+					.setDefaultRequestConfig(defaultRequestConfig).disableCookieManagement();
+
+			if (myProxy != null && StringUtils.isNotBlank(getProxyUsername()) && StringUtils.isNotBlank(getProxyPassword())) {
+				CredentialsProvider credsProvider = new BasicCredentialsProvider();
+				credsProvider.setCredentials(new AuthScope(myProxy.getHostName(), myProxy.getPort()),
+						new UsernamePasswordCredentials(getProxyUsername(), getProxyPassword()));
+				builder.setProxyAuthenticationStrategy(new ProxyAuthenticationStrategy());
+				builder.setDefaultCredentialsProvider(credsProvider);
+			}
+
+			myHttpClient = builder.build();
+
+		}
+
+		return myHttpClient;
+	}
+
+	@Override
+	protected void resetHttpClient() {
+		this.myHttpClient = null;
+	}
+
+	/**
+	 * Only allows to set an instance of type org.apache.http.client.HttpClient
+	 * @see ca.uhn.fhir.rest.client.api.IRestfulClientFactory#setHttpClient(Object)
+	 */
+	@Override
+	public synchronized void setHttpClient(Object theHttpClient) {
+		this.myHttpClient = (HttpClient) theHttpClient;
+	}
+
+	@Override
+	public void setProxy(String theHost, Integer thePort) {
+		if (theHost != null) {
+			myProxy = new HttpHost(theHost, thePort, "http");
+		} else {
+			myProxy = null;
+		}
+	}
+
+	private PoolingHttpClientConnectionManager createInsecureHttpConnectionManager(int timeToLive, TimeUnit timeUnit) {
+		X509TrustManager trustManager = new X509TrustManager() {
```

Comment:
this seems `instance`-able, or do you need a new one every time for thread stuff?

Reply:
That should be doable.